app/app.go

```diff
@@ -776,6 +775,69 @@ func Delete(ctx context.Context, app *App, evt *event.Event, requestID string) e
 	return nil
 }
 
+// DeleteVersion deletes an app version.
+func (app *App) DeleteVersion(ctx context.Context, w io.Writer, versionStr string) error {
+	w = app.withLogWriter(w)
+	msg := fmt.Sprintf("\n ---> Deleting version %s of app %s", versionStr, app.Name)
+	fmt.Fprintf(w, "%s\n", msg)
+	var hasErrors bool
+	defer func() {
+		var problems string
+		if hasErrors {
+			problems = " Some errors occurred during removal."
+		}
+		fmt.Fprintf(w, "---- Done removing application version %s.%s\n", versionStr, problems)
+	}()
+
+	logErr := func(msg string, err error) {
+		msg = fmt.Sprintf("%s: %s", msg, err)
+		fmt.Fprintf(w, "%s\n", msg)
+		log.Errorf("[delete-app-version: %s-%s] %s", app.Name, versionStr, msg)
+		hasErrors = true
+	}
+
+	_, version, err := app.explicitVersion(versionStr)
+	if err != nil {
+		return err
+	}
+	prov, err := app.getProvisioner()
+	if err != nil {
+		return err
+	}
+
+	versionInt, err := strconv.Atoi(versionStr)
+	if err != nil {
+		return err
+	}
+
+	if cleanProv, ok := prov.(provision.CleanImageProvisioner); ok {
```

Comment:
Unique provider that implements CleanImageProvisioner is the dockercluster. Could you remove this part